test/e2e/apps/deployment.go

```diff
@@ -1476,3 +1480,75 @@ func waitForDeploymentCompleteAndCheckRolling(c clientset.Interface, d *appsv1.D
+	deploy, err := c.AppsV1().Deployments(ns).Create(context.TODO(), d, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	// Wait for it to be updated to revision 1
+	err = e2edeployment.WaitForDeploymentRevisionAndImage(c, ns, deploymentName, "1", WebserverImage)
+	framework.ExpectNoError(err)
+
+	err = e2edeployment.WaitForDeploymentComplete(c, deploy)
+	framework.ExpectNoError(err)
+
+	deployment, err := c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	newRS, err := deploymentutil.GetNewReplicaSet(deployment, c.AppsV1())
+	framework.ExpectNoError(err)
+	framework.ExpectNotEqual(newRS, nilRs)
+
+	ginkgo.By("getting scale subresource")
+	scale, err := c.AppsV1().Deployments(ns).GetScale(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		framework.Failf("Failed to get scale subresource: %v", err)
+	}
+	framework.ExpectEqual(scale.Spec.Replicas, int32(1), "")
+	framework.ExpectEqual(scale.Status.Replicas, int32(1), "")
+
+	ginkgo.By("updating a scale subresource")
+	scale.ResourceVersion = "" // indicate the scale update should be unconditional
+	scale.Spec.Replicas = 2
+	scaleResult, err := c.AppsV1().Deployments(ns).UpdateScale(context.TODO(), deploymentName, scale, metav1.UpdateOptions{})
+	if err != nil {
+		framework.Failf("Failed to put scale subresource: %v", err)
+	}
+	framework.ExpectEqual(scaleResult.Spec.Replicas, int32(2), "")
+
+	ginkgo.By("verifying the deployment Spec.Replicas was modified")
+	deployment, err = c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		framework.Failf("Failed to get deployment resource: %v", err)
+	}
+	framework.ExpectEqual(*(deployment.Spec.Replicas), int32(2), "")
+
+	ginkgo.By("Patch a scale subresource")
+	scale.ResourceVersion = "" // indicate the scale update should be unconditional
+	scale.Spec.Replicas = 4    // should be 2 after "UpdateScale" operation, now Patch to 4
+	deploymentScalePatchPayload, err := json.Marshal(autoscalingv1.Scale{
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: scale.Spec.Replicas,
+		},
+	})
+	framework.ExpectNoError(err, "Could not Marshal JSON for patch payload")
+
+	_, err = c.AppsV1().Deployments(ns).Patch(context.TODO(), deploymentName, types.StrategicMergePatchType, []byte(deploymentScalePatchPayload), metav1.PatchOptions{}, "scale")
+	framework.ExpectNoError(err, "Failed to patch deployment: %v", err)
+
+	ginkgo.By("verifying the deployment Spec.Replicas was modified")
+	deployment, err = c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	framework.ExpectNoError(err, "Failed to get deployment resource: %v", err)
+	framework.ExpectEqual(*(deployment.Spec.Replicas), int32(4), "deployment should have 4 replicas")
+	e2edeployment.WaitForDeploymentComplete(c, deployment)
```

Comment:
Also not needed, as with the other PRs, you only care about update+patch operation to succeed. Scaling itself is tested elsewhere.

Reply:
Delete complete block

Reply:
Think I understood wrong. Replace `get' lines and only removed `WaitForDeploymentComplete`